src/components/catalog-filter/catalog-filter.controller.ts

```diff
@@ -0,0 +1,174 @@
+import * as angular from 'angular';
+import * as _ from 'lodash';
+
+export class CatalogFilterController implements angular.IController {
+  static $inject = ['$scope', 'Catalog'];
+
+  public ctrl: any = this;
+  private $scope: any;
+  private Catalog: any;
+  private removeClearFilterListener: any;
+
+  constructor($scope: any, Catalog: any) {
+    this.$scope = $scope;
+    this.Catalog = Catalog;
+  }
+
+  public $onInit() {
+    this.ctrl.filterPanelModel = [
+      {
+        id: 'keyword',
+        title:  'Keyword',
+        placeholder: 'Filter by Keyword',
+        filterType: 'text',
+        values: []
+      }
+    ];
+    this.ctrl.keywordFilter = this.ctrl.filterPanelModel[0];
```

Comment:
Seems a bit backwards to me. Better to create the keyword filter object, then push it to the filterPanelModel.

Reply:
Fixed